lib/engine/step/g_1873/buy_sell_par_shares.rb

```diff
@@ -0,0 +1,118 @@
+# frozen_string_literal: true
+
+require_relative '../buy_sell_par_shares'
+require_relative '../share_buying'
+require_relative '../../action/buy_shares'
+require_relative '../../action/par'
+
+module Engine
+  module Step
+    module G1873
+      class BuySellParShares < BuySellParShares
+        def description
+          'Sell then Buy Certificates or Form Public Mine'
+        end
+
+        def purchasable_companies(_entity)
+          []
+        end
+
+        # FIXME: need to deal with receivership first and second buy
+        def can_buy_multiple?(_entity, corporation)
+          return false unless @game.railway?(corporation)
+
+          @current_actions.any? { |x| x.is_a?(Action::Par) && x.corporation == corporation } &&
+            @current_actions.none? { |x| x.is_a?(Action::BuyShares) }
+        end
+
+        def can_sell?(entity, bundle)
+          return unless bundle
+
+          corporation = bundle.corporation
+
+          timing = @game.check_sale_timing(entity, corporation)
+
+          timing &&
+            !(@game.class::MUST_SELL_IN_BLOCKS && @round.players_sold[entity][corporation] == :now) &&
+            can_sell_order? &&
+            (@game.share_pool.fit_in_bank?(bundle) || corporation == @game.mhe) &&
+            can_dump?(entity, bundle) &&
+            president_can_sell?(entity, corporation)
+        end
+
+        # president of corp can't dump unless someone else has 20% - even if president only has 10%
+        # this may be the only reason we keep track of the presidents share
+        def can_dump?(entity, bundle)
+          corporation = bundle.corporation
+
+          return true unless bundle.presidents_share
+          return true if corporation == @game.mhe
+          return false if !corporation.operated? && @game.railway?(corporation)
+
+          sh = corporation.player_share_holders(corporate: true)
+          (sh.reject { |k, _| k == entity }.values.max || 0) >= 20
+        end
+
+        # president of RR can never drop below 20% if it hasn't operated
+        def president_can_sell?(entity, corporation)
+          return true unless corporation.owner == entity
+          return true if !@game.railway?(corporation) || corporation.operated? || corporation == @game.mhe
+
+          corporation.share_holders[entity] > 20
+        end
+
+        def ipo_type(entity)
+          if @game.railway?(entity)
+            :par
+          else
+            :form
+          end
+        end
+
+        def get_par_prices(entity, corp)
+          super if @game.railway?(corp)
```

Comment:
does super have a side effect? you don't use it anywhere did you mean return super?

Reply:
This was a half-completed thought. No need for super at all.